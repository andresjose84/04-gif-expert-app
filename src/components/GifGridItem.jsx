import React from 'react'
import PropTypes from 'prop-types';
const GifGridItem = ({id,title,url} ) => {
    return (
        <div className='card animate__animated animate__bounce' key={id}>
            <img src={url} alt={`img-${id}`} title={title}/>
            <p>{title}</p>
        </div>
    )
}
GifGridItem.propTypes = {
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
export default GifGridItem;