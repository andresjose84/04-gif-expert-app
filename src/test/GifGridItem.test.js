import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../components/GifGridItem';
describe('Pruebas GifGridItem', () => { 
    test('snapshot test', () => { 
        const obj = {
            id:1,
            title:'test',
            url:'https://localhost:3000/imagen'
        }
        const wrapper  = shallow( <GifGridItem { ...obj }/>);
        expect(wrapper).toMatchSnapshot();
     });
 });