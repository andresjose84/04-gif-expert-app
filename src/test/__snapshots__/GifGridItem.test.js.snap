// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridItem snapshot test 1`] = `
<div
  className="card animate__animated animate__bounce"
  key="1"
>
  <img
    alt="img-1"
    src="https://localhost:3000/imagen"
    title="test"
  />
  <p>
    test
  </p>
</div>
`;
